lib/manifest/EndListAppender.js

```diff
@@ -24,31 +25,40 @@ var EndlistAppender = function EndListAppender(){
         var that = this;
-        return Q.fcall(function () {
-            var entryFolder = persistenceFormat.getEntryFullPath(entryId);
-            var getManifestFiles = Q.denodeify(glob);
-            return getManifestFiles("**/*.m3u8", {cwd: entryFolder, realpath: true});
-        }).then(function (flavorManifestPaths) {
-            var promises = _.map(flavorManifestPaths, function (flavorManifestPath) {
-                return m3u8Parser.parseM3U8(flavorManifestPath)
-                    .then(function (manifest) {
-                        logger.debug("[%s] Adding ENDLIST to: %s", entryId, flavorManifestPath);
-                        if (manifest.get('playlistType') === 'VOD') {
-                            return;
-                        }
-                        manifest.set('playlistType', 'VOD');
-                        return qio.write(flavorManifestPath, manifest.toString());
-                    })
-                    .catch(function (err) {
-                        logger.error("[%s] Adding EXT-X-ENDLIST at location: %s failed; Error: %s", entryId, flavorManifestPath, ErrorUtils.error2string(err));
-                    });
+
+        backendClient.getLiveEntryServerNodes(entryId)
+            .then( function(liveEntryServerNodes) {
+                if (liveEntryServerNodes.count > 0) {
+                    logger.debug("EntryId %s, has at least one providing data center", entryId);
+                    return Q.reject();
+                }
+            })
+            .catch( function () {
+                logger.error(" failed to retrieve server nodes providing EntryId %s", entryId);
+                return Q.reject();
+
+            })
+            return Q.fcall(function () {
+                    var entryFolder = persistenceFormat.getEntryFullPath(entryId);
+                    var getManifestFiles = Q.denodeify(glob);
+                    return getManifestFiles("**/*.m3u8", {cwd: entryFolder, realpath: true});
+            }).then(function (flavorManifestPaths) {
+                var promises = _.map(flavorManifestPaths, function (flavorManifestPath) {
+                    return m3u8Parser.parseM3U8(flavorManifestPath)
+                        .then(function (manifest) {
+                            logger.debug("[%s] Adding ENDLIST to: %s", entryId, flavorManifestPath);
+                            if (manifest.get('playlistType') === 'VOD') {
+                                return;
+                            }
+                            manifest.set('playlistType', 'VOD');
+                            return qio.write(flavorManifestPath, manifest.toString());
+                        })
+                        .catch(function (err) {
+                            logger.error("[%s] Adding EXT-X-ENDLIST at location: %s failed; Error: %s", entryId, flavorManifestPath, ErrorUtils.error2string(err));
+                        });
+                });
+
             });
```

Comment:
what about the catch retriggering the timeout again? otherwise this function will never be called again